cachito/workers/tasks/gomod.py

```diff
@@ -49,4 +50,10 @@ def fetch_gomod_source(request_id, dep_replacements=None):
     }
     env_vars.update(config.cachito_default_environment_variables.get("gomod", {}))
     update_request_with_package(request_id, module, env_vars)
-    update_request_with_deps(request_id, module, deps)
+    update_request_with_deps(request_id, module, module_deps)
+
+    # add package deps
+    [p.pop("from_module") for p in [pkg] + pkg_deps]
+    update_request_with_package(request_id, pkg)
+    update_request_with_deps(request_id, pkg, pkg_deps)
+    update_request_with_deps(request_id, pkg, module_deps)
```

Comment:
Rather than associate the `module_deps` with the package, I think it'd make more sense to somehow associate the package with the module. This would reduce duplication in the database for every request and speed up the request as the dependencies wouldn't have to be sent a second time to the PATCH API endpoint, which checks to see if each dependency already exists in the database. This would save a couple minutes to a large module.